src/app/admin/invite/invite-form.tsx

```diff
@@ -0,0 +1,99 @@
+'use client'
+
+import { TextInput, PasswordInput, Button, Input, Checkbox, Text, Flex } from '@mantine/core'
+import { useMutation } from '@tanstack/react-query'
+import { useForm } from '@mantine/form'
+import { notifications } from '@mantine/notifications'
+import { OrganizationSelect } from './organization-select'
+import Link from 'next/link'
+import { adminInviteUserAction } from './admin-users.actions'
+import { InviteUserFormValues, inviteUserSchema, zodResolver } from './admin-users.schema'
+import { randomString } from '@/lib/string'
+import { reportError } from '@/components/errors'
+
+export function InviteForm() {
+    const studyProposalForm = useForm({
+        mode: 'uncontrolled',
+        validate: zodResolver(inviteUserSchema),
+        validateInputOnBlur: true,
+        initialValues: {
+            firstName: '',
+            lastName: '',
+            email: '',
+            organizationId: '',
+            password: randomString(8),
```

Comment:
commented in the `lib/string` about using faker to generate the string here if we want to